rclpy/src/rclpy/_rclpy_logging.c

```diff
@@ -73,6 +75,7 @@ rclpy_logging_get_logger_severity_threshold(PyObject * Py_UNUSED(self), PyObject
     PyErr_Format(PyExc_RuntimeError,
       "Failed to get severity threshold for logger \"%s\", return code: %d\n",
       name, severity);
+    rcutils_reset_error();
```

Comment:
Note: today the `rcutils_logging_get_logger_severity_threshold` doesn't set an error message itself but since it/something it calls might do so in the future, and FWIU [rcutils_reset_error can be called even without an error](https://github.com/ros2/rcutils/blob/002c03c188771cc43405fd4e66cc8052d1eeb91e/src/error_handling.c#L292), I chose to leave this.